src/Sound/AudioDebug.cpp

```diff
@@ -0,0 +1,287 @@
+		ImGui::Text("%s", emitter.world ? "Yes" : "No");
+		ImGui::NextColumn();
+		if (emitter.world)
+			ImGui::Text("(%.1f,%.1f, %.1f)", emitter.position.x, emitter.position.y, emitter.position.z);
+		else
+			ImGui::Text("N/A");
+		ImGui::NextColumn();
+	}
+	ImGui::EndChild();
+	ImGui::SameLine();
+	ImGui::BeginChild("Audio Decoder Settings", ImVec2(500, ImGui::GetContentRegionAvail().y), true);
+	ImGui::Separator();
+	ImGui::Text("Audio Decoder settings");
+	if (ImGui::Button(("Re-initialize Audio Decoder (" + handler.GetAudioDecoderName() + ")").c_str()))
+		handler.ReplaceAudioLoader(std::make_unique<FfmpegDecoder>(FfmpegDecoder()));
+	ImGui::Separator();
+	ImGui::EndChild();
+	ImGui::SameLine();
+	ImGui::BeginChild("Audio Player Settings", ImGui::GetContentRegionAvail(), true);
+	ImGui::Text("Audio Player settings");
+	if (ImGui::Button(("Re-initialize Audio Player (" + handler.GetAudioPlayerName() + ")").c_str()))
+		handler.ReplaceAudioPlayer(std::make_unique<OpenAlPlayer>());
+	ImGui::Separator();
+	float volume = handler.GetGlobalVolume();
+	ImGui::SliderFloat("Global Volume", &volume, 0.0f, 1.0f, "%.3f");
+	ImGui::SliderFloat("Music Volume", &volume, 0.0f, 1.0f, "%.3f");
+	ImGui::SliderFloat("SFX Volume", &volume, 0.0f, 1.0f, "%.3f");
+	ImGui::Separator();
+	if (volume != handler.GetGlobalVolume())
+		handler.SetGlobalVolume(volume);
+	ImGui::Text("Active Sounds");
+	ImGui::SameLine();
+	if (ImGui::Button("Play") && _selectedEmitter != InitialAudioEmitterId)
+		handler.PlayEmitter(_selectedEmitter);
+	ImGui::SameLine();
+	if (ImGui::Button("Pause") && _selectedEmitter != InitialAudioEmitterId)
+		handler.PauseEmitter(_selectedEmitter);
+	ImGui::SameLine();
+	if (ImGui::Button("Stop") && _selectedEmitter != InitialAudioEmitterId)
+		handler.StopEmitter(_selectedEmitter);
+	ImGui::SameLine();
+	if (ImGui::Button("Export to .wav") && _selectedEmitter != InitialAudioEmitterId)
+		AudioDecoder::DebugEmitWavFile(*handler.GetSound(handler.GetEmitters().at(_selectedEmitter).soundId));
+	ImGui::Separator();
+	ImGui::Columns(6, "PlayingSounds", true);
+	ImGui::Text("Audio Source ID");
+	ImGui::NextColumn();
+	ImGui::Text("Audio Buffer ID");
+	ImGui::NextColumn();
+	ImGui::Text("Sound Name");
+	ImGui::NextColumn();
+	ImGui::Text("Buffer size");
+	ImGui::NextColumn();
+	ImGui::Text("Status");
+	ImGui::NextColumn();
+	ImGui::Text("Progress");
+	ImGui::NextColumn();
+	ImGui::Separator();
+	for (auto [emitterId, emitter] : emitters)
+	{
+		if (ImGui::Selectable(("##" + std::to_string(emitterId)).c_str(), _selectedEmitter == emitterId,
+		                      ImGuiSelectableFlags_SpanAllColumns))
+			_selectedEmitter = emitterId;
+		ImGui::SameLine();
+		ImGui::Text("%u", emitter.audioSourceId);
+		ImGui::NextColumn();
+		ImGui::Text("%u", emitter.audioBufferId);
+		ImGui::NextColumn();
+		ImGui::Text("%s", handler.GetSound(emitter.soundId)->name.c_str());
+		ImGui::NextColumn();
+		ImGui::Text("%llu", handler.GetSound(emitter.soundId)->bytes.size());
```

Comment:
error: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 2 has type ‘long unsigned int’ [-Werror=format=]